pg_search/src/index/directory/mvcc.rs

```diff
@@ -63,6 +63,11 @@ pub struct MVCCDirectory {
     // cloned, we don't lose all the work we did originally creating the FileHandler impls.  And
     // it's cloned a lot!
     readers: Arc<Mutex<FxHashMap<PathBuf, Arc<dyn FileHandle>>>>,
+
+    // a lazily loaded [`IndexMeta`], which is only created once per MVCCDirectory instance
+    // we cannot tolerate tantivy calling `load_metas()` multiple times and giving it a different
+    // answer
+    loaded_metas: OnceLock<tantivy::Result<IndexMeta>>,
```

Comment:
This, along with the change to the `load_metas()` function below, is the fix.

Everything else is code that was necessary to track it down, so I've included it as part of this PR.